### PR TITLE
Support separate entropy target for nested actions

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/utils/dist_utils.py [hunk 1]
```diff
@@ -680,6 +680,8 @@ def entropy_with_fallback(distributions):
     Args:
         distributions (nested Distribution): A possibly batched tuple of
             distributions.
+        return_sum (bool): if return_sum, return the total entropy. If not return_sum,
```
Comment (hnyu) [inline alf/utils/dist_utils.py]: if True ... if not True

File: alf/algorithms/entropy_target_algorithm.py [hunk 2]
```diff
@@ -32,7 +33,18 @@
...
+                           skip_free_stage=False,
+                           max_entropy=None,
+                           target_entropy=None,
+                           very_slow_update_rate=0.001,
+                           slow_update_rate=0.01,
+                           fast_update_rate=np.log(2),
+                           min_alpha=1e-4,
+                           average_window=2,
+                           debug_summaries=False,
+                           name="EntropyTargetAlgorithm"):
```
Comment (hnyu) [inline alf/algorithms/entropy_target_algorithm.py:47]: Should add an explanation that _NestedEntropyTargetAlgorithm will be created for nested action_spec.

File: alf/algorithms/entropy_target_algorithm.py [hunk 3]
```diff
@@ -32,7 +33,18 @@
 
 
 @alf.configurable
-class EntropyTargetAlgorithm(Algorithm):
+def EntropyTargetAlgorithm(action_spec,
```
Comment (le-horizon) [inline alf/algorithms/entropy_target_algorithm.py]: this is no longer an Algorithm, maybe call it entropy_target_algorithm_ctor?

**Review by le-horizon — COMMENTED** [review]
Almost LG.  Just one name nit.

Thanks,
Le

File: alf/algorithms/entropy_target_algorithm.py (hunk 3 above)
Comment (le-horizon) [reply alf/algorithms/entropy_target_algorithm.py]: > this is no longer an Algorithm, maybe call it entropy_target_algorithm_ctor?

@emailweixu other way of creating a separate NestedEntropyTargetAlgorithm class, keeping the current EntropyTargetAlgorithm unchanged also looks good to me.

**Review by hnyu — APPROVED** [review]

**Pull request merged**